From 2bea913c33990828cfe8fe59371e5d7a62f17f55 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Tue, 26 Apr 2022 06:06:10 +0200
Subject: [PATCH] arm32le: only use cache sysconf constant if they are defined

This fixes compilation on musl libc which presently does not implement
these sysconf constants [0].

[0]: https://www.openwall.com/lists/musl/2017/07/24/1
---
 c/arm32le.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/c/arm32le.c b/c/arm32le.c
index e27242b..bf09cfa 100644
--- a/c/arm32le.c
+++ b/c/arm32le.c
@@ -41,11 +41,13 @@ void S_doflush(uptr start, uptr end) {
 void S_machine_init(void) {
   int l1_dcache_line_size, l1_icache_line_size;
 
-  if ((l1_dcache_line_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) <= 0) {
+#ifdef _SC_LEVEL1_DCACHE_LINESIZE
+  if ((l1_dcache_line_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE)) <= 0)
+#endif
     l1_dcache_line_size = DEFAULT_L1_MAX_CACHE_LINE_SIZE;
-  }
-  if ((l1_icache_line_size = sysconf(_SC_LEVEL1_ICACHE_LINESIZE)) <= 0) {
+#ifdef _SC_LEVEL1_ICACHE_LINESIZE
+  if ((l1_icache_line_size = sysconf(_SC_LEVEL1_ICACHE_LINESIZE)) <= 0)
+#endif
     l1_icache_line_size = DEFAULT_L1_MAX_CACHE_LINE_SIZE;
-  }
   l1_max_cache_line_size = l1_dcache_line_size > l1_icache_line_size ? l1_dcache_line_size : l1_icache_line_size;
 }
